### PR TITLE
Authorization on GET generic events

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Related Issue(s)
- #295

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
  - [x] GET for org
  ~~- [ ] GET For end user: not applicable~~ out of scope
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
~~- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)~~

### CODE REVIEW

File: src/Events/Authorization/GenericCloudEventXacmlMapper.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        /// </summary>
+        internal static List<XacmlJsonCategory> CreateMultipleResourceCategory(List<CloudEvent> events)
+        {
+            List<XacmlJsonCategory> resourcesCategories = new();
+            int counter = 1;
+
+            foreach (CloudEvent cloudEvent in events)
+            {
+                var resourceCategory = CreateResourceCategory(cloudEvent);
+                resourceCategory.Id = CloudEventXacmlMapper.ResourceId + counter.ToString();
```
Comment (b-brodie-digdir) [inline src/Events/Authorization/GenericCloudEventXacmlMapper.cs:40]: Would it look nice with a separator char here? Something that keeps the original ResourceId distinguisable.

File: test/Altinn.Platform.Events.Tests/TestingControllers/EventsControllerTests.cs [hunk 2]
```diff
@@ -64,8 +62,11 @@ public EventsControllerTests(WebApplicationFactory<EventsController> factory)
                     Id = Guid.NewGuid().ToString(),
                     Type = "system.event.occurred",
                     Subject = "/person/16069412345",
-                    Source = new Uri("urn:isbn:1234567890"),
+                    Source = new Uri("urn:isbn:1234567890")
                 };
+
+                _validEvent["resource"] = "urn:altinn:rr:nbib.bokoversikt.api";
```
Comment (b-brodie-digdir) [inline test/Altinn.Platform.Events.Tests/TestingControllers/EventsControllerTests.cs:68]: Did we agree on "urn:altinn:rr" or "urn:altinn:resoure"? I thought I was sure, but now I'm not.

File: src/Events/Controllers/EventsController.cs [hunk 3]
```diff
@@ -147,6 +144,17 @@ private static (bool IsValid, string ErrorMessage) ValidateQueryParams(string af
             return (true, null);
         }
 
+        private static (bool IsValid, string ErrorMessage) ValidateCloudEvent(CloudEvent cloudEvent)
+        {
+            string eventResource = cloudEvent["resource"].ToString();
```
Comment (acn-sbuad) [inline src/Events/Controllers/EventsController.cs:149]: could wait with this until we implement authorization of POST. But added it during discussions. Can remove if needed.

File: src/Events/Authorization/GenericCloudEventXacmlMapper.cs (hunk 1 above)
Comment (acn-sbuad) [reply src/Events/Authorization/GenericCloudEventXacmlMapper.cs:40]: Resource id is simply an `R`, so the result will be R1, R2 for instance. I believe the category Id is easier to read without a seperator

File: test/Altinn.Platform.Events.Tests/TestingControllers/EventsControllerTests.cs (hunk 2 above)
Comment (acn-sbuad) [reply test/Altinn.Platform.Events.Tests/TestingControllers/EventsControllerTests.cs:68]: Not quite sure about that. Given that its only testdata and the value does affect the functionalty I didn't research. resource is correct we can update.

**Pull request merged**